src/main/java/uk/gov/hmcts/reform/laubackend/idam/controllers/UserDeletionAuditController.java

```diff
@@ -168,7 +173,63 @@ public ResponseEntity<UserDeletionGetResponse> getUserDeletions(
             );
             appInsights.trackEvent(
                 GET_DELETED_ACCOUNTS_INVALID_REQUEST_EXCEPTION.toString(),
-                appInsights.trackingMap("exception", ire.getMessage())
+                appInsights.trackingMap(EXCEPTION, ire.getMessage())
+            );
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
+
+    }
+
+    @Operation(
+        tags = "User Accounts endpoints",
+        summary = "Retrieve all deleted user accounts",
+        description = "Query all deleted user accounts based on page, size and sort"
+    )
+    @ApiResponses({
+        @ApiResponse(
+            responseCode = "200",
+            description = "Request executed successfully. Response will contain list of deleted users",
+            content = { @Content(schema = @Schema(implementation = UserDeletionGetResponse.class))}),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Missing required parameter(s). page and size are both required ",
+            content = { @Content(schema = @Schema(implementation = UserDeletionGetResponse.class))}),
+        @ApiResponse(
+            responseCode = "403",
+            description = "Forbidden",
+            content = { @Content(schema = @Schema(implementation = UserDeletionGetResponse.class))}),
+        @ApiResponse(
+            responseCode = "500",
+            description = "Internal Server Error",
+            content = { @Content(schema = @Schema(implementation = UserDeletionGetResponse.class))})
+    })
+    @GetMapping(
+        path = "/audit/getAllDeletedAccounts",
+        produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseBody
+    public ResponseEntity<UserDeletionGetResponse> getAllDeletedUsers(
+        @Parameter(name = "Authorization", example = "Bearer eyJ0eXYiOiJK.........")
+        @RequestHeader(value = AUTHORISATION_HEADER) String authToken,
+        @Parameter(name = "Service Authorization", example = "Bearer eyJ0eXYiOiJK.........")
+        @RequestHeader(value = SERVICE_AUTHORISATION_HEADER) String serviceAuthToken,
+        DeletionLogAllUsersRequestParams requestParams
+    ) {
+        try {
+            verifyAllUserDeletionGetRequestParams(requestParams);
```

Comment:
We should also check that if sort is given, it is one of "asc" or "desc" and not something else.